src/block_cache/block_chunkcache.c

```diff
@@ -153,12 +199,13 @@ __chunkcache_free_chunk(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
     if (chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY)
         __wt_free(session, chunk->chunk_memory);
     else {
-#ifdef ENABLE_MEMKIND
-        memkind_free(chunkcache->memkind, chunk->chunk_memory);
-#else
-        __wt_err(session, EINVAL,
-          "Chunk cache requires libmemkind, unless it is configured to be in DRAM");
-#endif
+        /* Update the bitmap to show free chunk, then free the memory of the chunk */
+        size_t index = (size_t)(chunk->chunk_memory - chunkcache->memory) / chunkcache->chunk_size;
+        do {
+            map_byte = (uint8_t *)&chunkcache->bitmap[index / 8];
+            ret = __wt_atomic_cas8(map_byte, *map_byte, *map_byte & ~(0x01 << (index % 8)));
+        } while (ret != 0);
```

Comment:
So, if I understand this correctly, multiple threads can attempt to free the bit. If one thread successfully completes the compare-and-swap operation, we would prefer the other threads to exit gracefully instead of attempting the compare-and-swap again.